src/studio/src/designer/backend/Factories/ModelFactory/JsonSchemaToMetamodelConverter.cs

```diff
@@ -542,384 +543,405 @@ private static string ConvertToCSharpCompatibleName(string name)
-        {
-            return $"{id} : [{minOccurs}..{maxOccurs}] {typeName}";
-        }
+                return null;
+            }
 
-        private static string GetDataBindingName(ElementType @type, int maxOccurs, string id, string fixedValue, string xPath)
-        {
-            if (@type != ElementType.Group && id.Contains(".") && string.IsNullOrEmpty(fixedValue))
+            private static ElementType GetType(JsonSchema subSchema)
             {
-                return GetDataBindingName(id, xPath);
+                var xsdAttributeTypeKeyword = subSchema.GetKeyword<XsdAttributeKeyword>();
+                if (xsdAttributeTypeKeyword?.Value == true)
+                {
+                    return ElementType.Attribute;
+                }
+
+                return ElementType.Field;
             }
-            else if (@type == ElementType.Group && maxOccurs > 1)
+
+            private static string GetDisplayString(string id, string typeName, int minOccurs, int maxOccurs)
             {
-                return GetDataBindingName(id, xPath);
+                return $"{id} : [{minOccurs}..{maxOccurs}] {typeName}";
             }
 
-            return null;
-        }
-
-        private static string GetDataBindingName(string id, string xPath)
-        {
-            var firstPropertyName = id[0..id.IndexOf(".")];
-            string dataBindingNameWithoutFirstPropertyName = xPath.Replace("/" + firstPropertyName + "/", string.Empty);
-            string dataBindingName = dataBindingNameWithoutFirstPropertyName.Replace("/", ".");
-
-            return dataBindingName;
-        }
-
-        private bool IsNillableType(JsonPointer path)
-        {
-            return _schemaXsdMetadata.GetCompatibleTypes(path).Contains(CompatibleXsdType.Nillable);
-        }
+            private static string GetDataBindingName(ElementType @type, int maxOccurs, string id, string fixedValue, string xPath)
+            {
+                if (@type != ElementType.Group && id.Contains(".") && string.IsNullOrEmpty(fixedValue))
+                {
+                    return GetDataBindingName(id, xPath);
+                }
+                else if (@type == ElementType.Group && maxOccurs > 1)
+                {
+                    return GetDataBindingName(id, xPath);
+                }
 
-        private bool IsRestrictionType(JsonPointer path)
-        {
-            return _schemaXsdMetadata.GetCompatibleTypes(path).Contains(CompatibleXsdType.SimpleTypeRestriction);
-        }
+                return null;
+            }
 
-        private static string GetTypeNameFromRefPath(JsonPointer pointer)
-        {
-            if (pointer.Segments.Length != 2 || (pointer.Segments[0].Value != "$defs" && pointer.Segments[0].Value != "definitions"))
+            private static string GetDataBindingName(string id, string xPath)
             {
-                return string.Empty;
-            }
+                var firstPropertyName = id[0..id.IndexOf(".")];
```

Comment:
What does this do? I'm not familiar with this syntax. (I can see that this was also present in the old code, just curious 😄 )

Reply:
It's the [range operator](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/ranges)
It will take substring from 0 to first occurence of "."